src/Semantic/Graph.hs

```diff
@@ -383,7 +384,7 @@ resumingResolutionError :: ( Has Trace sig m
 resumingResolutionError = runResolutionErrorWith $ \ baseError -> do
   traceError "ResolutionError" baseError
   case baseErrorException baseError of
-    NotFoundError nameToResolve _ _ -> pure $ Path.absRel nameToResolve
+    NotFoundError nameToResolve _ _ -> pure $ fromMaybe (Path.toAbsRel Path.emptyFile) $ Path.fileFromFileDir nameToResolve
```

Comment:
The empty path isn’t strictly correct here, but I’m not sure of the ramifications of that. For now, can you call `trace` or `traceError` here with information about this name that isn’t structurally valid?

Reply:
I think the best way to fix it is to change the definition of ResolutionError as 
```
data ResolutionError resume where
  NotFoundError :: Path.AbsRelFileDir   -- The path that was not found.
                -> [Path.AbsRelFile] -- List of paths searched that shows where semantic looked for this module.
                -> Language -- Language.
                -> ResolutionError Path.AbsRelDir -- which is ModulePath currently.
``` 

If you agree with this fix, I will make a TODO here and fix it in next PR or this PR.

Reply:
I think, respectively, `GoImportError` shall be typed as 

```
GoImportError :: String -> ResolutionError [String]
```

Reply:
Hi, this is about something in pathtype.  The only possibility of "Path.fileFromFileDir nameToResolve" is that the nameToResolve is empty, like `Path.currentDir`.

I have done some tests in ghci, which shows some subtle differences 
```
> Path.fileFromFileDir $ Path.toFileDir $ Path.currentDir 
Nothing
> Path.fileFromFileDir $ Path.toFileDir $ Path.relDir "./"
Just (currentDir </> relPath ".")
> Path.fileFromFileDir $ Path.toFileDir $ Path.relDir "."
Just (currentDir </> relPath ".")
```

Reply:
I think the problem is that we want to resume with a ModulePath anyway.  Perhaps, in the future, we can think about define module path with status.
``
  data Status = Resolved | NotFound | Undecided
  data ModulePath (x :: Status ) where
      Resolved :: (ResolvedOrUndecided x) Path.AbsRelFile -> ModulePath x
      NotFound :: (NotFoundOrUndecided x) String -> ModulePath x
```

Then we can resume with `ModulePath 'Undecided`, and process blob with `ModulePath 'Resolved`.  And then make a simple adapter between these two.

I am not sure whether this design may indicate a fused-effect library.

Reply:
A traceError temporary fix is added by c2542a1